src/platform_impl/web/device/mod.rs

```diff
@@ -0,0 +1,114 @@
+mod gamepad;
+
+use super::event_loop::EventLoop;
+use crate::event::device;
+pub use gamepad::GamepadShared;
+use std::{
+    cmp::{Eq, Ordering, PartialEq, PartialOrd},
+    hash::{Hash, Hasher},
+};
+
+macro_rules! device_id {
```

Comment:
I think creating a macro here is overkill: it decrease readability for a small decrease in LoC.

Reply:
Well I can split. I took example from the window plateform implementation.